pkg/agent/upstream/remote/remote.go

```diff
@@ -118,7 +120,11 @@ func (r *Remote) uploadProfile(j *upstream.UploadJob) error {
 
 	r.Logger.Debugf("uploading at %s", u.String())
 	// new a request for the job
-	request, err := http.NewRequest("POST", u.String(), bytes.NewReader(j.Trie.Bytes()))
+	ce, body, err := r.compressBody(j.Trie.Bytes())
```

Comment:
General go comment:

I think something like [this](https://gist.github.com/tomcatzh/cf8040820962e0f8c04700eb3b2f26be) would be slightly better because it would use less memory due to streaming, but I'm not 100% sure maybe there are downsides / other considerations I'm missing.

@kolesnikovae what do you think?

Reply:
A gzip reader fits perfectly, I think. Also, can't find any downsides, but the example snippet may need some better error handling 

BTW, if `Trie` satisfied `io.WriterTo`, we'd be able to make the streaming even more efficient:
```go
func NewGzipReader(src io.WriteTo) io.Reader
```